### PR TITLE
feat(forms): Add prettyValue function for form toasts

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Makes form toasts a decent bit better, and less "engineering" feeling.

- `true` and `false` are now displayed as "enabled" and "disabled"
- `null` renders as "<none>"
- `undefined` renders as "<unset>"

A `formatMessageValue` can now be specified on a fields properties (or through the field configuration) which can be declared as a function to format the message, or as false to just disable showing the changed values.

Adds formatters for:

 - Select dropdown, looksup the value from the choices
 - Range slider, uses the `formatLabel` function
 - Choice Mapper disables showing the changes.

### CODE REVIEW

**Comment by EvanPurkhiser** [comment]
Ready for review @billyvg

File: src/sentry/static/sentry/app/actionCreators/indicator.jsx [hunk 1]
```diff
@@ -69,12 +87,20 @@ export function saveOnBlurUndoMessage(change, model, fieldName) {
 
   if (!label) return;
 
+  let prettifyValue = val => prettyFormString(val, model, fieldName);
+  let showChangeText = model.getDescriptor(fieldName, 'formatMessageValue') !== false;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/actionCreators/indicator.jsx]: I would add a comment here saying we only hide change text when `formatMessageValue` is explicitly `false`.

File: src/sentry/static/sentry/app/views/settings/components/forms/selectField.jsx [hunk 2]
```diff
@@ -23,6 +33,8 @@ export default class SelectField extends React.Component {
     escapeMarkup: true,
     multiple: false,
     small: false,
+    formatMessageValue: (value, props) =>
+      (getChoices(props).find(v => v[0] === value) || [null, value])[1],
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/selectField.jsx]: Would prefer either if `v` has a more descriptive name (e.g. `choice`) or destructure e.g. `.find(([choiceValue]) => choiceValue === value)`

**Review by billyvg — APPROVED** [review]
Left small notes but this looks great, thanks for doing this!

**Pull request merged**